src/app/core/auth/validators/emailValidator.ts

```diff
@@ -0,0 +1,24 @@
+import { AbstractControl, AsyncValidatorFn, ValidationErrors } from '@angular/forms'
+import { catchError, delay, map, Observable, of } from 'rxjs'
+import { switchMap } from 'rxjs/operators'
+import { AuthService } from '@core/auth/services/auth.service'
+import { HttpErrorResponse } from '@angular/common/http'
+
+export function emailValidator(authService: AuthService): AsyncValidatorFn {
+  return (control: AbstractControl): Observable<ValidationErrors | null> => {
+    if (!control.value) {
+      return of(null)
+    }
```

Comment:
AbstractControl receives generic type of the value, provide it so you don't need to cast